arsteinm/safety-app
# US-31: User signup page

# Overview
As the project sponsor, I want new users to input basic user information, so that I can identify which users from which schools completed the walkthroughs.

# Description
Created the user signup page, which falls between the welcome pages and the walkthroughs landing page. If the user has been through the welcome pages, but hasn't completed the sign up, then on next use it will drop them directly on the sign up page. If they have completed the sign up, then on next use it will drop them directly on the walkthroughs landing page.

The form includes simple validation to make sure all fields are completed and that the email at least has an '@' in it. Didn't feel the need to go too crazy on validation for the MVP.

When saving the school and user to the db, it defaults to setting the schoolId and userId to 1, since the signup form will show up for completion once. We may have to take a different approach when syncing the data with the remote db, otherwise we'll have a whole of users with the same id.

# How to test
Once you complete and save the form, you can verify that it saved to the database by querying the user and schools tables in `sqlite3` via `adb`, using the following entries in the Android Studio terminal or from the command line (step 1 assumes that your `Android/sdk/platform-tools` directory is on your system PATH):

1. `adb shell`
2. `su`
3. `cd /data/data/com.plusmobileapps.safetyapp/databases`
4. `sqlite3`
5. `.open appDB.db`
6. `SELECT * FROM user;`
7. `SELECT * FROM schools;`
8. ...or any select statement you prefer if you want to practice your SQL skills. Optionally, before step 6, you can enter `.headers on` if you want to see the field name headers in your results.

Or you can `pull` the database into a local directory through `adb` and inspect it in your favorite SQLite GUI browser, but that takes about the same number of steps.

# Screenshot
Add some screenshots of your code, if applicable.
![image](https://user-images.githubusercontent.com/8429746/36431072-e0a76676-1613-11e8-9255-85c9b015fa63.png) ![image](https://user-images.githubusercontent.com/8429746/36431168-21b561c2-1614-11e8-90dd-d1ebc5cbcdcd.png)

# Trello Link
https://trello.com/c/dqHefJZ9

# Merge Checklist 
[X ] Backmerged master in your branch.
[n/a] All unit tests built and passed, or TODO with linked stories if unit tests
    have failed.
